services/api/src/lib/utils/schema.js

```diff
@@ -57,3 +63,46 @@ function isDisallowedField(doc, key, allowPrivate = false) {
   }
   return false;
 }
+
+exports.loadModel = (definition, name) => {
+  const { attributes } = definition;
+  if (!attributes) {
+    throw new Error(`Invalid model definition for ${name}, need attributes`);
+  }
+  const schema = exports.createSchema(attributes);
+  schema.plugin(require('mongoose-autopopulate'));
+  return mongoose.model(name, schema);
+};
+
+exports.loadModelDir = (dirPath) => {
+  const files = fs.readdirSync(dirPath);
+  for (const file of files) {
+    const basename = path.basename(file, '.json');
+    const basenameJs = path.basename(file, '.js');
+    if (file.match(/\.js$/) && !files.includes(`${basenameJs}.json`)) {
+      console.warn(`Found a model ${basenameJs}.js but no corresponding ${basenameJs}.json`);
+    }
+    if (file.match(/\.js$/)) {
+      const modelName = startCase(basenameJs).replace(/\s/g, '');
+      if (!mongoose.models[modelName]) {
```

Comment:
even if we don't do the above, I don't think this part is necessary as there's already 2 guards to prevent mongoose models from being double loaded, one being the `module.exports = mongoose.models.User || mongoose.model('User', schema);` line at the end of every model and also CJS requires themselves which have a cache mechanism